cloud/google/clusteractuator.go

```diff
@@ -22,28 +22,40 @@ import (
 	"golang.org/x/net/context"
 	"golang.org/x/oauth2/google"
 	"google.golang.org/api/compute/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/record"
+
 	"sigs.k8s.io/cluster-api/cloud/google/clients"
 	"sigs.k8s.io/cluster-api/cloud/google/clients/errors"
 	gceconfigv1 "sigs.k8s.io/cluster-api/cloud/google/gceproviderconfig/v1alpha1"
 	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 	client "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/typed/cluster/v1alpha1"
+	apierrors "sigs.k8s.io/cluster-api/pkg/errors"
 )
 
 const (
 	firewallRuleAnnotationPrefix = "gce.clusterapi.k8s.io/firewall"
 	firewallRuleInternalSuffix   = "-allow-cluster-internal"
 	firewallRuleApiSuffix        = "-allow-api-public"
+
+	createClusterEventAction = "Create"
+	deleteClusterEventAction = "Delete"
 )
 
 type GCEClusterClient struct {
 	computeService         GCEClientComputeService
 	clusterClient          client.ClusterInterface
 	gceProviderConfigCodec *gceconfigv1.GCEProviderConfigCodec
+	eventRecorder          record.EventRecorder
+	v1Alpha1Client         client.ClusterV1alpha1Interface
```

Comment:
clusterClient is specfic to clusters in the default namespace
and v1alpha1client is generic but used for clusters.
can you combine these two